### PR TITLE
GHA: trigger CSI tests on go.mod update

### PR DESCRIPTION
**Pull request by isinyaaa**
This prevents CSI dependencies from being out-of-date.

## Merge criteria:

- [ ] The commits and have meaningful messages; the author will squash them [after approval](https://github.com/opendatahub-io/opendatahub-community/blob/main/contributor-cheatsheet.md#:~:text=Usually%20this%20is%20done%20in%20last%20phase%20of%20a%20PR%20revision) or will ask to merge with squash.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: .github/workflows/csi-test.yml [hunk 1]
```diff
@@ -4,18 +4,20 @@ on:
...
+      - ".github/ISSUE_TEMPLATE/**"
+      - ".github/dependabot.yml"
+      - "docs/**"
   pull_request:
     paths:
       - "csi/**"
       - "internal/server/openapi/api_model_registry_service*"
       - "pkg/openapi/**"
+      # csi build depends on base go.mod https://github.com/kubeflow/model-registry/issues/311
+      - "go.mod"
```
Comment (tarilabs) [inline .github/workflows/csi-test.yml:20]: @lampajr please be informed, as apparently there is some "further dependency ties" which we discovered (more in the linked issue)

**Review by tarilabs — COMMENTED** [review]
/lgtm

**Comment by tarilabs** [comment]
many thanks Isabella, will wait for another to go in then proceeding to merge this one

File: .github/workflows/csi-test.yml (hunk 1 above)
Comment (lampajr) [reply .github/workflows/csi-test.yml:20]: Good catch!!!

I think that makes totally sense, as the CSI test actually makes use of the local (thus relative) model registry dependency.. therefore the root go.mod is used (see https://github.com/kubeflow/model-registry/blob/d2312907025adbe83d3faafbecf1474824d055ed/csi/Dockerfile.dev#L23).

**Review by lampajr — COMMENTED** [review]
/lgtm

**Review by tarilabs — APPROVED** [review]

**Pull request merged**